dask/dataframe/io/tests/test_parquet.py

```diff
@@ -4557,6 +4557,47 @@ def test_pyarrow_filesystem_option(tmp_path, fs):
     assert_eq(ddf, df)
 
 
+@PYARROW_MARK
+def test_fsspec_to_parquet_filesystem_option(tmp_path):
+    from fsspec import get_filesystem_class
+    from pyarrow import dataset as pa_da
+
+    key1 = "/read1"
+    key2 = str(tmp_path / "write1")
+
+    df = pd.DataFrame({"a": range(10)})
+    fs = get_filesystem_class("memory")(use_instance_cache=False)
+    pa_da.write_dataset(
+        pa.Table.from_pandas(df),
+        key1,
+        format="parquet",
+        filesystem=fs,
+    )
```

Comment:
```suggestion

    key1 = "/read1"
    key2 = str(tmp_path / "write1")

    df = pd.DataFrame({"a": range(10)})
    fs = get_filesystem_class("memory")(use_instance_cache=False)
    df.to_parquet(key1, engine="pyarrow", filesystem=fs)
```

Reply:
Pretty sure pandas will pass through the `filesystem` option for you.

Reply:
I like the ability to skip instance cache, so I chose it. I think I could have used `memory:///{key1}`.

Reply:
Didn't know about this, cool!